linux/LinuxProcessList.c

```diff
@@ -1323,6 +1341,15 @@ static bool LinuxProcessList_recurseProcTree(LinuxProcessList* this, openat_arg_
       return false;
    }
 
+   bool autogroup = (settings->flags & PROCESS_FLAG_LINUX_AUTOGROUP);
+   if (autogroup) {
+      // Refer to sched(7) 'autogroup feature' section
+      // The kernel feature can be enabled/disabled through procfs at
+      // any time, so check for it at the start of each sample - only
+      // read from per-process procfs files if it's globally enabled.
+      autogroup = LinuxProcess_isAutogroupEnabled();
```

Comment:
This leads to `LinuxProcess_isAutogroupEnabled()` being called for each process and thread if `PROCESS_FLAG_LINUX_AUTOGROUP` is set.